cp-61: profile settings sidebar

vladyslavdobychin opened the pull request:
Setup basic layout for sidebar, but got a couple of [questions](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/issues/61#issuecomment-1693277338)

vladyslavdobychin commented:
Currently working on the conflicts.

v0ldemar01 commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-button/profile-settings-button.tsx:
```diff
@@ -0,0 +1,20 @@
+  const { icon, children } = properties;
+  return (
+    <button className={styles['profile-button']}>
+      <img src={icon} alt="button icon" />
```
Let's create an Icon component(check miniproject for some details) but not use FontAwesome, just render .svg

HIMDanny commented:
hey! I've noticed you deleted a lot of files that are used. What was the purpose?)

dmytro-kucherenko commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-button/profile-settings-button.tsx:
```diff
@@ -0,0 +1,21 @@
+const ProfileSettingsButton: React.FC<ProfileSettingsButtonProperties> = (
+  properties,
+) => {
+  const { icon, children } = properties;
```
You can detruct your proprs already in line 9

dmytro-kucherenko commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-button/styles.module.scss:
```diff
@@ -0,0 +1,27 @@
+  cursor: pointer;
+}
+
+.profile-button > img {
```
You can't use global tags in styles, B10 rule

dmytro-kucherenko commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-button/styles.module.scss:
```diff
@@ -0,0 +1,27 @@
+  color: var(--gray-300);
+  font-weight: 600;
+  font-size: 16px;
+  font-family: Montserrat, sans-serif;
```
I think it's better to set basic font-family in body element in scaffolding to not set it every time in other places, but there's still no merged pull request with this approach

dmytro-kucherenko replied:
We already have default font family in body element, so you can remove it)

fshabanov replied:
Resolve this comment

dmytro-kucherenko commented on frontend/src/libs/components/profile-settings-sidebar/styles.module.scss:
```diff
@@ -0,0 +1,59 @@
+.container {
+  max-width: 349px;
+  min-height: 100vh;
+  background: linear-gradient(180deg, rgb(229 240 255 / 0%) 0%, #e5f0ff 100%);
```
You should use constants frrom variables with only hex, not rgb

vladyslavdobychin replied:
Well, that' the tricky part, in some cases, an element in Figma would have a hex that we do not have in our variables

dmytro-kucherenko replied:
In this case you can add new one, all variables are there because someone was needed them)

vladyslavdobychin replied:
That makes sense, just did not really want to mess with global variables until I get some sort of approval

Dmytro-Kucherenko reviewed (commented):
You maybe accidentally attached your issue #61 to the pr and it will be automatically closed after merging instead of being transfered to testing tab)
Also as I understand you created this branch from your local development branch without many new brand changes and at now there're lots of outdated files, that should be pulled

BrianSammit commented:
> hey! I've noticed you deleted a lot of files that are used. What was the purpose?)

I think the deletions were for the files that are in conflict.

fshabanov commented:
Please, don't forget to add screenshots

v0ldemar01 commented on frontend/src/libs/components/icon/icons/icon-map.tsx:
```diff
@@ -0,0 +1,17 @@
+import { ReactComponent as Plus } from './plus.svg';
+import { ReactComponent as SignOut } from './sign-out.svg';
+
+const iconNameToPlainSvgMap: Record<
+  ValueOf<typeof IconNameToIcon>,
+  JSX.Element
+> = {
+  [IconNameToIcon.ARROW]: <Arrow />,
+  [IconNameToIcon.PLUS]: <Plus />,
+  [IconNameToIcon.SIGN_OUT]: <SignOut />,
+};
```
Use union instead of enum

v0ldemar01 commented on frontend/src/libs/components/icon/icons/arrow.svg:
Let's move to assets/icons

v0ldemar01 commented on frontend/src/libs/components/icon/icons/icon-map.tsx:
```diff
@@ -0,0 +1,17 @@
+  ValueOf<typeof IconNameToIcon>,
+  JSX.Element
+> = {
+  [IconNameToIcon.ARROW]: <Arrow />,
+  [IconNameToIcon.PLUS]: <Plus />,
+  [IconNameToIcon.SIGN_OUT]: <SignOut />,
```
```suggestion
  [IconNameToIcon.ARROW]: ArrowIcon,
  [IconNameToIcon.PLUS]: PlusIcon,
  [IconNameToIcon.SIGN_OUT]: SignOutIcon,
```

Ekit18 replied:
@v0ldemar01 when i change my icon-map to this i get error
![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/92929504/1da0b297-b7cd-4331-8255-e01fb5998500)

vladyslavdobychin replied:
I also got a bit lost with this part

vladyslavdobychin replied:
Maybe it's: 

`[IconNameToIcon.ARROW]: <ArrowIcon />,
  [IconNameToIcon.PLUS]: <PlusIcon />,
  [IconNameToIcon.SIGN_OUT]: <SignOutIcon />,`

OlenaYatseniuk replied:
It's weird, but I don't have this console error after renaming.

v0ldemar01 commented on frontend/src/libs/components/chat-sidebar/chat-element/chat-element.tsx:
```diff
@@ -0,0 +1,47 @@
+      onClick={handleSelectChat}
+    >
+      <div className={styles['name']}>
+        <img src="chat-logo.svg" alt="not found" />
```
What about this icon?

vladyslavdobychin replied:
Working on it as we speak

v0ldemar01 replied:
Don't forget about this icon

vladyslavdobychin replied:
Sorry, I missed it. 
I also wanted to ask: 
Do I have edit it myself? As far as I remember, chat-element was Yvhenii's ticket that he's been working on. This component was just showed here since I was pulling his Icon component. Not sure If I won't create conflicts messing with it.

v0ldemar01 commented on frontend/src/libs/components/chat-sidebar/chat-element/chat-element.tsx:
```diff
@@ -0,0 +1,47 @@
+  id: 1,
+};
+
+const ChatElement: React.FC<Properties> = ({ chat }) => {
```
Maybe ChatItem would be more suitable here

vladyslavdobychin replied:
I cherry picked these changes from cp-55 alongside with the Icon component I needed, Yevhenii has been working on it

v0ldemar01 replied:
We need to stay with only one version, please check it to prevent duplication of code after merging

vladyslavdobychin replied:
Is it fine if I directly copy files from cp-55 (where the chat element is initially developed) to my branch using git checkout?

vladyslavdobychin replied:
Mister @v0ldemar01 I just got an idea: 
chat-element.tsx was cherrypicked from cp-55, it's by any means not integrated to my branch and not used anywhere. Can I just delete and it won't be causing conflicts when Yevhenii will merging his PR?

v0ldemar01 commented on frontend/vite.config.ts:
```diff
@@ -13,7 +14,11 @@ const config = ({ mode }: ConfigEnv): ReturnType<typeof defineConfig> => {
+    plugins: [
+      reactPlugin(),
+      vitePWAPlugin({ registerType: 'autoUpdate' }),
+      svgr(),
```
Please rename as the existing plugins

vladyslavdobychin replied:
Will just pulling the changes from development work? These are the changes I also picked up from cp-55 trying to get the Icon component

v0ldemar01 replied:
Got it, but as that changes are in your PR, you need to improve it related to styles or to remove it. Check if correct naming is in latest development

vladyslavdobychin replied:
I did, what I found is that in development there are identical plugins except svgr(), I have it my branch since it's needed for the Icon component. At least, this is how initially Yevhenii Kitan set it up. I am not sure what I need to change in here.

HIMDanny replied:
I guess @v0ldemar01 means it should be named like `svgrPlugin`

v0ldemar01 commented on frontend/src/libs/components/icon/icon.tsx:
```diff
@@ -0,0 +1,14 @@
+};
+
+const Icon: React.FC<Properties> = ({ name }) => (
+  <>{iconNameToPlainSvgMap[name]}</>
```
@Ekit18 that is the reason of that console error. The logic should be something like this:
```jsx
const SelectedIcon = iconNameToIcon[name];

return <SelectedIcon className={getValidClasses(styles.icon, className)} />;
```

v0ldemar01 commented on frontend/src/libs/components/icon/icon-map.tsx:
Let's move to libs/maps/maps and follow with full name of file icon-name-to-plain-svg.map.ts

v0ldemar01 commented on frontend/src/libs/components/icon/icon-map.tsx:
```diff
@@ -0,0 +1,22 @@
+  ValueOf<typeof IconNameToIcon>,
+  React.FunctionComponent<React.SVGProps<SVGSVGElement>>
+> = {
+  [IconNameToIcon.ARROW]: ArrowIcon,
```
Use union instead of enum

vladyslavdobychin replied:
Sorry, missed that, will look into it

fshabanov commented on frontend/src/assets/icons/subscription.svg:
I suggest we download only the icons in the middle and add background ourselves. In these downloaded versions it feels like the icons are not centered well

vladyslavdobychin replied:
True, I'll do my best.

fshabanov commented on frontend/src/libs/components/icon/icon.tsx:
```diff
@@ -0,0 +1,16 @@
+import { type ValueOf } from '#/libs/types/types.js';
```
```suggestion
import { type ValueOf } from '#libs/types/types.js';
```

vladyslavdobychin commented:
Hi, @v0ldemar01 , could you please have a look at the [latest commit](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/pull/75/commits/9ccea8dcdc0ab8060e3ab472e0f63f9843f03758) where I tried replacing IconNameToIcon enum with a union type, but I'm not sure If I'm going in the right direction.

vladyslavdobychin commented:
Adjusted the icon component to also apply css classes for icons in the profile settings sidebar that require background. 
All svgs in the assets/icon were adjusted to have the same sizing. 
The only issue is notifications icon, which seems to be a bit not aligned by itself, not sure if it needs fixing since it wasn't in my issue initially and I added it purely for testing purposes, to see if multiple buttons will be populated  without problems. 
Currently working on adding shadows for the active option.

OlenaYatseniuk commented on frontend/src/libs/components/icon/icon.tsx:
```diff
@@ -0,0 +1,23 @@
+const Icon: React.FC<Properties> = ({ name, className = '' }) => {
+  const SelectedIcon = iconNameToPlainSvgMap[name];
+
+  const cssClass = className && styles[className] ? styles[className] : '';
+
```
I think it's better to use getValidClassNames for this purpose. It could be something like this.
```suggestion
  const classNames = getValidClassNames(className, styles['default-className']);

```

fshabanov commented on frontend/src/assets/css/variables.scss:
```diff
@@ -2,13 +2,15 @@
   --white: #ffffff;
   --shadow: 0 2px 4px #0000001a;
   --black: #000000;
+  --gray-50: #ffffff80;
```
We already have this color. --white-alpha-80 below

fshabanov commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-button/profile-settings-button.tsx:
```diff
@@ -0,0 +1,40 @@
+import { useCallback } from 'react';
```
Import from #libs/hooks/hooks.js

fshabanov commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-sidebar.tsx:
```diff
@@ -0,0 +1,59 @@
+import { useCallback, useState } from 'react';
```
Import from hooks file

fshabanov commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-sidebar.tsx:
```diff
@@ -0,0 +1,59 @@
+        </div>
+        <div className={styles['buttons-container']}>
+          <ProfileSettingsButton
+            name={'notification'}
```
```suggestion
            name="notification"
```

fshabanov commented on frontend/src/libs/components/profile-settings-sidebar/styles.module.scss:
```diff
@@ -0,0 +1,64 @@
+  max-width: 349px;
+  min-height: 100vh;
+
+  // For the second variable in the linear-gradient function the --blue-100 value is used as a replacement for #e5f0ff, if the slight difference is not acceptable, then the --blue-50 could be created with the value #e5f0ff
```
Remove all comments

fshabanov commented on frontend/src/libs/maps/icon-name-to-plain-svg.tsx:
```diff
@@ -0,0 +1,21 @@
+  IconNames,
+  React.FunctionComponent<React.SVGProps<SVGSVGElement>>
+> = {
+  'arrow': ArrowIcon,
+  'avatar': AvatarIcon,
+  'plus': PlusIcon,
+  'sign-out': SignOutIcon,
+  'notification': NotificationIcon,
+  'subscription': SubscriptionIcon,
```
Do we need quotes here?

Xxandder replied:
I think yes, because we can't use 'sign-out' without quotes

HIMDanny replied:
yeah, but quotes are only needed for 'sign-out', aren't they?

vladyslavdobychin replied:
so it should be all without quotes and only one with quotes? May look messy doesn't it?

dmytro-kucherenko replied:
If there's already one have to be with quotes, I think it's better to stick to the same style with all of them, but not sure

v0ldemar01 replied:
This map should be in icon package

fshabanov commented on package.json:
```diff
@@ -49,5 +49,8 @@
     "commit-msg": "npx commitlint --edit $1"
+  },
+  "dependencies": {
+    "vite-plugin-svgr": "3.2.0"
```
I think it should be in frontend folder

v0ldemar01 replied:
Not resolved yet

fshabanov commented on shared/src/libs/enums/icon-name.enum.ts:
```diff
@@ -0,0 +1,18 @@
+// const IconNameToIcon = {
+//   ARROW: 'arrow',
+//   PLUS: 'plus',
+//   SIGN_OUT: 'sign-out',
+//   AVATAR: 'avatar',
+//   NOTIFICATION: 'notification',
+//   SUBSCRIPTION: 'subscription',
+// } as const;
```
```suggestion
```

Xxandder commented on shared/src/libs/enums/icon-name.enum.ts:
```diff
@@ -0,0 +1,18 @@
+//   SUBSCRIPTION: 'subscription',
+// } as const;
+
+type IconNames =
```
Name of the type should be the same as file name. And you should put type in types folder, not enums

Xxandder commented on shared/src/index.ts:
```diff
@@ -5,6 +5,7 @@ export {
   ExceptionMessage,
   ServerErrorType,
 } from './libs/enums/enums.js';
+export { type IconNames } from './libs/enums/icon-name.enum.js';
```
export { type IconNames } from './libs/enums/enums.js';

Xxandder commented on frontend/src/libs/components/profile-settings-sidebar/styles.module.scss:
```diff
@@ -0,0 +1,64 @@
+.container {
+  max-width: 349px;
+  min-height: 100vh;
```
I'm not sure, but I think we should avoid using vh

HIMDanny replied:
we should, as Liza said, it may cause issues on mobile

Xxandder commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-sidebar.tsx:
```diff
@@ -0,0 +1,59 @@
+          <div className="visually-hidden">User details</div>
+          <div className={styles['user-details']}>
+            <div className={styles['user-icon']}>
+              <Icon name={'avatar'} />
```
name = "avatart" (based on this [comment](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/pull/75/files#r1311837713))

dmytro-kucherenko replied:
And also without curly braces, only string

Xxandder commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-sidebar.tsx:
```diff
@@ -0,0 +1,59 @@
+  return (
+    <div className={styles['container']}>
+      <div className={styles['header']}>
+        <div className="visually-hidden">Profile settings</div>
```
@fshabanov Do we need here import scaffolding.scss and use styles _ile['class_name'] notation?

vladyslavdobychin replied:
Well, I just followed the general approach for other components that were merged before

HIMDanny replied:
we don't need to import `scaffolding.scss` as it is available globally

BrianSammit reviewed (dismissed):
With the final changes requested should be good to go!! 💪

juanc-escobar commented:
I was going through your latest update in the code and looks pretty good, the code is well organized the definitions of the types look good, the naming seems consistent, and imports are readable, I think you did a great job, you just need to fix the details in the comments and it should be ready for merge. 🎉

v0ldemar01 commented on frontend/src/libs/components/icon/icon.tsx:
```diff
@@ -0,0 +1,23 @@
+  const cssClass = className && styles[className] ? styles[className] : '';
+
+  return (
+    <div className={cssClass}>
```
Remove div wrapping, move className to Icon via props

v0ldemar01 commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-button/profile-settings-button.tsx:
```diff
@@ -0,0 +1,40 @@
+
+  return (
+    <button className={buttonStyle} onClick={handleOnClick}>
+      <Icon name={name} className={'profile-button-background'} />
```
How is it working properly?

vladyslavdobychin replied:
Not sure how but it does 

https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/8e609586-15d5-4c28-9d26-62d663b68ee7

vladyslavdobychin replied:
<img width="497" alt="Screenshot 2023-09-01 at 12 55 32" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/d128e2e9-caf0-4aa0-b74d-36f832ca4c28">

v0ldemar01 replied:
it cannot work properly.

vladyslavdobychin commented:
Current setup, keep working on comments 


https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/21093184-cd99-4e4e-960c-c8d4767ace78

v0ldemar01 commented on shared/src/libs/enums/icon-names.type.ts:
```diff
@@ -0,0 +1,9 @@
+type IconNames =
```
```suggestion
type IconName =
```

vladyslavdobychin commented:
https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/06aa9801-6289-4a33-a6f6-9ba65f72046a

dmytro-kucherenko commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-button/profile-settings-button.tsx:
```diff
@@ -0,0 +1,41 @@
+
+import styles from './styles.module.scss';
+
+type ProfileSettingsButtonProperties = {
```
Lets use `Properties` instead because we already get used to it

dmytro-kucherenko commented on frontend/src/libs/enums/enums.ts:
```diff
@@ -6,3 +6,4 @@ export {
   ContentType,
   ServerErrorType,
 } from 'shared/build/index.js';
+export { type IconName } from 'shared/build/index.js';
```
It's already import from that path above, just add `IconName` there

dmytro-kucherenko commented on frontend/src/libs/components/icon/icon.tsx:
```diff
@@ -0,0 +1,20 @@
+const Icon: React.FC<Properties> = ({ name, className = '' }) => {
+  const SelectedIcon = iconNameToPlainSvgMap[name];
+
+  const cssClass = getValidClassNames(styles[className]);
```
From one side you pass `className` prop in order to set styles outside. From other side this class name is used in `styles` object, so you limit yourself only via styles in range of imported scss file.

I think there're 2 ways to make it more consistent: or you receive `className: classValue` (from lib) and pass the whole `style['smth']` outside, or just make prop `className: 'profile-button-background' | 'further-option'` and use these kind of presets from imported file. The second seems better to me

fshabanov replied:
Let's not pass classes or styles to Icon component. For background color, margin, padding we can wrap Icon component on div where we use it. If we need to change the color of icon, we can pass `color` property

v0ldemar01 replied:
@vladyslavdobychin please follow Farid's comment, it's mandatory one

liza-veis commented on frontend/src/assets/icons/notification.svg:
remove images that are not used by your pr

fshabanov commented on frontend/src/assets/css/variables.scss:
```diff
@@ -9,6 +9,7 @@
   --blue-100: #ebf3ff;
   --blue-200: #acb4d0;
   --blue-300: #507ceb;
+  --blue-400: #3295cd52;
```
Use `blue-alpha-(value)` format

fshabanov commented on frontend/src/libs/components/icon/icon.tsx:
```diff
@@ -0,0 +1,20 @@
+import { type IconName } from '#libs/enums/enums.js';
+import { getValidClassNames } from '#libs/helpers/get-valid-class-names.js';
```
```suggestion
import { getValidClassNames } from '#libs/helpers/get-valid-class-names.js';
```
```suggestion
import { getValidClassNames } from '#libs/helpers/helpers.js';
```

fshabanov commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-button/profile-settings-button.tsx:
```diff
@@ -0,0 +1,41 @@
+import { Icon } from '#libs/components/components.js';
+import { type IconName } from '#libs/enums/enums.js';
+import { getValidClassNames } from '#libs/helpers/get-valid-class-names.js';
```
```suggestion
import { getValidClassNames } from '#libs/helpers/helpers.js';
```

Xxandder replied:
@fshabanov Do you mean that we can pass color property to an icon component or you said about that we can set coolor for icon in css in wrapper element?

fshabanov replied:
@Xxandder we can pass `color` property to Icon component. I don't think it would be possible to change icon color using wrapper

Xxandder replied:
@fshabanov We can set color with css styles. it was disscussed [here](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/pull/140#discussion_r1312694559). So we set fill="currentColor" in svg and this svg will have color as in parent element

fshabanov commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-button/profile-settings-button.tsx:
```diff
@@ -0,0 +1,41 @@
+  onClick,
+  isActive = false,
+}) => {
+  const handleOnClick = useCallback(() => {
```
```suggestion
  const handleClick = useCallback(() => {
```

v0ldemar01 commented on shared/src/libs/enums/icon-name.type.ts:
```diff
@@ -0,0 +1,9 @@
+type IconName =
```
It is called union, not enum. Rename it to types

v0ldemar01 replied:
Why is this item in the shared folder?

v0ldemar01 reviewed (changes requested):
Please re-request after finishing working on comments

vladyslavdobychin commented:
@v0ldemar01 seems done to me, I also tried my best implementing Farid's comment. Seems to be working just fine: 

https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/d28d8874-705b-4478-ba33-1218446cc5a9

v0ldemar01 commented on frontend/src/libs/components/icon/icon-name-to-plain-svg.tsx:
Not sure if the extension of .tsx is needed here

v0ldemar01 reviewed (changes requested):
LGTM, please check last comments

fshabanov commented on frontend/src/assets/css/variables.scss:
```diff
@@ -16,6 +16,7 @@
   --transparent: #00000000;
   --white-alpha-20: #ffffff33;
   --white-alpha-80: #ffffff80;
+  --blue-alpha-90: #3295cd52;
```
Let's keep an order here, put --blue-alpha-90 after --blue-alpha-70

fshabanov reviewed (changes requested):
How does the layout look on mobile device? Could you share some screenshots?

Xxandder commented on frontend/src/libs/components/icon/icon.tsx:
```diff
@@ -0,0 +1,14 @@
+import { iconNameToPlainSvgMap } from '#libs/components/icon/icon-name-to-plain-svg.js';
```
I think you can use relative path here bacause this mapper is in icon package

Xxandder commented on frontend/src/libs/components/icon/icon-name-to-plain-svg.ts:
```diff
@@ -0,0 +1,23 @@
+import { ReactComponent as SubscriptionIcon } from '#assets/icons/subscription.svg';
+import { type IconName } from '#libs/types/types.js';
+
+const iconNameToPlainSvgMap: Record<
```
rename according to this [comment](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/pull/140) and cp-55 pr

vladyslavdobychin replied:
Which comment? It's the link to the entire PR

Xxandder replied:
Oops. Here is the [link](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/pull/140#discussion_r1312696725) Also it's names IconNameToIconin current development branch

Xxandder commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-button/styles.module.scss:
```diff
@@ -0,0 +1,45 @@
+  font-weight: 600;
+  font-size: 16px;
+  font-family: Montserrat, sans-serif;
+  line-height: 20px;
```
use here absolute values: 1,25

vladyslavdobychin replied:
EMs or REMs?

Xxandder replied:
just 1.2 https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/pull/140#discussion_r1312692998

Xxandder commented on frontend/src/libs/components/icon/icon-name-to-plain-svg.ts:
rename file according to a returned value name

Xxandder commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-button/styles.module.scss:
```diff
@@ -0,0 +1,45 @@
+
+.profile-button-text {
+  display: block;
+  width: 145px;
```
it's better to avoid using fixed width and height values if it's possible

vladyslavdobychin replied:
I might need to keep it, I needed the text not to break trough the specific line, I'll try to see what I can do

liza-veis replied:
could you add screenshots on what you are trying to achieve, what If text exceeds this line? looks like in this case it should be moved to the next line and we don't need width for this

Xxandder commented on frontend/src/libs/components/profile-settings-sidebar/styles.module.scss:
```diff
@@ -0,0 +1,64 @@
+  font-weight: 600;
+  font-size: 16px;
+  font-family: Montserrat, sans-serif;
+  line-height: 20px;
```
same for line-heigh there

vladyslavdobychin commented:
@fshabanov there are no mobile styles defined for the component. I was not accounting for it (it was not really stated in the ticket + I was not sure that this component will be even used on mobile) 
Should I start adding mobile styles? 

<img width="555" alt="Screenshot 2023-09-05 at 12 44 10" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/318ee77f-712d-433a-afe3-5f0d5e6fb3ea">
<img width="545" alt="Screenshot 2023-09-05 at 12 44 21" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/c1bfe82c-bfb8-4619-b2ab-3d6d9efb958a">

fshabanov commented:
@vladyslavdobychin Every component should be mobile responsive and it shouldn't be mentioned in each ticket.
For mobile sizes, together with other coaches we decided to have similar screens like in mobile application. This sidebar component should be like a drawer. Please, take a look at mobile implementation and do the similar implementation

vladyslavdobychin commented:
Current draft of mobile responsiveness 


https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/e16837f3-3f92-40fd-adb6-966c865d138d

vladyslavdobychin commented on frontend/src/index.tsx:
This change was done purely for testing purposes, it will be removed in the subsequent commits

vladyslavdobychin commented:
@fshabanov @v0ldemar01 could you please have a look?

v0ldemar01 commented on shared/src/libs/types/icon-name.type.ts:
```diff
@@ -0,0 +1,9 @@
+type IconName =
```
It still in shared

vladyslavdobychin replied:
frontend/libs/types in this case?

v0ldemar01 reviewed (changes requested):
And rename file with icon mapper

liza-veis commented on frontend/src/libs/components/icon/icon-name-to-plain-svg.ts:
```diff
@@ -0,0 +1,21 @@
+import { ReactComponent as ArrowIcon } from '#assets/icons/arrow.svg';
```
we should already have this mapper, you can remove this file and move your icons to existing one

liza-veis commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-button/profile-settings-button.tsx:
```diff
@@ -0,0 +1,44 @@
+  isActive?: boolean;
+};
+
+const ProfileSettingsButton: React.FC<Properties> = ({
```
Can't we reuse Card component for this?

vladyslavdobychin replied:
There was not card component initially but I will see how to utilise it right now

vladyslavdobychin replied:
@Liza-Veis 
I went over the Card component trying to figure out how to implement it instead of the ProfileSettingsButton component. Right now, I can't say that I could just apply it and roll with it. 
Card component uses img and does not utilise the Icon component we developed, it does not account for an Icon to have a background either, it also does not use the arrow icon for marking active button, it does not account for the additional icon at all (Primary icon of the button + arrow icon or trash bin icon) 
But I do like the idea of having this card universal for all components. If you tell me to, I can work on the Card component to make it account for all use cases. I'm just worried about the fact that it may significantly postpone the merge of my PR.

liza-veis replied:
let's add iconName property to Card and if it's passed while image not, render icon instead

liza-veis commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-sidebar.tsx:
```diff
@@ -0,0 +1,66 @@
+      <div className={styles['body']}>
+        <div className={styles['buttons-container']}>
+          <div className="visually-hidden">Profile settings options</div>
+          <ProfileSettingsButton
+            name="notification"
+            onClick={handleOnClick}
+            isActive={activeButton === 'notification'}
+          >
+            Very long text to see if it will break
+          </ProfileSettingsButton>
+
+          <ProfileSettingsButton
+            name="subscription"
+            onClick={handleOnClick}
+            isActive={activeButton === 'subscription'}
+          >
+            Test
+          </ProfileSettingsButton>
+
```
remove test data
```suggestion
```

vladyslavdobychin replied:
I will, of course, remove them. I just kept them until the PR is completely approved to make sure I can always test the entire functionality without need for adding stuff back and forth.

v0ldemar01 commented on frontend/src/libs/maps/icon-name-to-plain-svg.ts:
It should be in frontend/src/libs/components/icon/libs/maps/icon-name-to-icon.map.ts

vladyslavdobychin replied:
My apologies, I think you excidently fused two paths. 
Just to confirm: frontend/src/libs/maps/icon-name-to-icon.map.ts
If so - already done.

v0ldemar01 replied:
No, this mapper is used only in icon component so we don't need to expose it to shared libs/maps/maps

vladyslavdobychin replied:
Okay, so I move it back to components/icon?

v0ldemar01 reviewed (changes requested):
Only one comment

vladyslavdobychin commented:
@Liza-Veis answering your [question](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/pull/75#discussion_r1316873342) : 
I gave the text of the button a fixed width to fix the arrow icon that marks the active button in one specific position, if there is no fixed width, the arrow icon will float closer to the text. 
This way I tried to ensure that text (as on figma) will always occupy the same space and the arrow Icon will always be in the same spot. 
<img width="424" alt="Screenshot 2023-09-06 at 14 46 48" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/efc570f7-03de-4d49-bc1d-ec7249a609d8">
<img width="412" alt="Screenshot 2023-09-06 at 14 47 15" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/e6e3d647-252e-4261-8eb7-d81913708528">

liza-veis commented:
@vladyslavdobychin use flex: 1 instead, you don't need fixed width, you need to expand this text, or align icon to the right

vladyslavdobychin commented:
@Liza-Veis I did some changes according to your comments regarding the fixed width, I used flex:1 but ran into a different issue - if the button is active it should render the arrow icon. If the text of the button is too long, it would be pushed away by the newly populated icon which created inconsistency in the design. 
I could not find a better solution to keep the gap between the arrow and the text and to render it more seamlessly rather than to move the arrow container outside of the statement that renders it. The only downside that bothers me is that there is an empty div, which will render the arrow icon when the button becomes active. 
Here are two videos before and after this solution. 
On the first video, the button renders the arrow container and arrow when you click on it, in the second the arrow container is always present and when you click on the button, it renders the arrow inside. 


<img width="626" alt="Screenshot 2023-09-06 at 17 35 20" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/323be413-9b63-4509-bd12-bc0c80d89fc1">

https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/59660092-e665-466b-9da5-1fe5f320fbeb

https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/bcf2ec90-7ef7-4e34-8c22-9e3f2fa50c89

fshabanov commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-sidebar.tsx:
```diff
@@ -0,0 +1,66 @@
+import { IconColor } from '#libs/enums/icon-color.enum.js';
```
```suggestion
import { IconColor } from '#libs/enums/enums.js';
```

fshabanov commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-sidebar.tsx:
```diff
@@ -0,0 +1,66 @@
+
+const ProfileSettingsSidebar: React.FC = () => {
+  const userName = useAppSelector((state) => {
+    return state.auth.authenticatedUser?.fullName;
```
Cast type here. We know that there is a user for sure, if the user reached this page

liza-veis commented:
@vladyslavdobychin about your question above - add arrow using absolute positioning, but reserve place for it using padding, it will be even better to add it in ::after

dmytro-kucherenko commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-sidebar.tsx:
```diff
@@ -0,0 +1,69 @@
+import styles from './styles.module.scss';
+
+const ProfileSettingsSidebar: React.FC = () => {
+  const userName = useAppSelector((state) => {
+    const user = state.auth.authenticatedUser as UserAuthResponseDto;
+
+    return user.fullName;
+  });
```
```suggestion
  const { userName } = useAppSelector((state) => {
    return {
       userName:  (state.auth.authenticatedUser as UserAuthResponseDto).fullName;
    }
  });
```
We just get used to this approach

Ekit18 replied:
may be it`s better to do smth like this:
  const fullName = useAppSelector(({ auth }) => {
    return (auth.authenticatedUser as UserAuthResponseDto).fullName;
  });

dmytro-kucherenko replied:
May be also an option, we just used object destruction in every app selector I saw, so I think it's more consistent way

OlesiaTr commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-button/styles.module.scss:
```diff
@@ -29,19 +40,16 @@
+  margin-left: 26px;
 }
 
 @media (width <= 480px) {
```
It should be (width < 480px), but I guess its one of the parts that you pulled from development

vladyslavdobychin commented:
Hello @Liza-Veis , I've been discussing with @Ekit18 on how to apply the Card component in my sidebar, I tried it and this is how it currently looks (screenshot below). 
I had to alter the Card component a bit, because Icon component inside of it needed a different wrapper div to render a correct background for (the blue box on the background). 
Also, in order to match the design the card component needs to display the additional Icon for the status (the arrow icon to show the active button) which it does not account for right now. 
If you say so, I can keep working on the Card component to add this functionality. 
Meanwhile I will be working on fixing other comments. 
Thanks for you attention. 

<img width="348" alt="Screenshot 2023-09-08 at 18 10 01" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/c58bfdda-4f47-4602-8f99-20acbd489931">
<img width="425" alt="Screenshot 2023-09-08 at 18 10 21" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/8907dd76-d61a-40a4-81f0-a92ce357e233">

fshabanov commented on frontend/src/libs/components/card/card.tsx:
```diff
@@ -10,6 +10,7 @@ type Properties = {
   onClick: () => void;
   isActive: boolean;
   iconName?: IconName;
+  iconColor?: string;
```
```suggestion
  iconColor?: ValueOf<IconColor>;
```

fshabanov commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-button/profile-settings-button.tsx:
```diff
@@ -0,0 +1,46 @@
+import { Icon } from '#libs/components/components.js';
+import { IconColor } from '#libs/enums/icon-color.enum.js';
```
```suggestion
import { IconColor } from '#libs/enums/enums.js';
```

vladyslavdobychin replied:
I see, my bad

fshabanov commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-sidebar.tsx:
```diff
@@ -0,0 +1,69 @@
+    return user.fullName;
+  });
+
+  const [activeButton, setActiveButton] = useState<IconName | ''>('');
```
```suggestion
  const [activeButton, setActiveButton] = useState<IconName | null>(null);
```

vladyslavdobychin replied:
Got it!

fshabanov commented on frontend/src/libs/components/profile-settings-sidebar/styles.module.scss:
```diff
@@ -0,0 +1,96 @@
+  padding: 0 16px;
+}
+
+@media (width <= 480px) {
```
```suggestion
@media (width < 480px) {
```
Same for others

vladyslavdobychin replied:
Understood, thank you.

vladyslavdobychin commented on frontend/src/libs/components/card/styles.module.scss:
```diff
@@ -28,10 +28,6 @@
 }
 
-.no-image {
-  padding-left: 20px;
```
the .no-image class initially caused profile settings sidebar icon to move away from the border: 
<img width="334" alt="Screenshot 2023-09-09 at 12 29 48" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/44b7c250-5d5b-46fd-ac57-f737e6a35284">

v0ldemar01 replied:
That changes are from another PR, check if that functionality working properly before removing it

vladyslavdobychin replied:
@v0ldemar01 I made sure I have latest changes from development and after removing this class I immediately checked if it affected other instances of the card component - no effect so far.

vladyslavdobychin replied:
You were correct, it is needed but for a component that was not in the development when I was making changes, I'm going to return in in the next commit

vladyslavdobychin replied:
@v0ldemar01 added back the .no-image style. 
I found the [conversation](https://bsa2023.slack.com/archives/C05P563E2MN/p1694184984693109)
where there was an issue with removing this class. 
I added back to the button styling and tested by removing icons and images from various buttons in two different sidebars. Hope will be fine.

https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/d8569e9f-18ee-4d33-bd50-29871fa7dc44

liza-veis commented:
@vladyslavdobychin

1. there should be padding around icon wrapper
2. there should be a reserved space for the arrow from the right side of the item (using padding), text should not cross this border
3. when text is wrapped on a new line, there should be no space on the left

<img width="344" alt="Screenshot 2023-09-09 at 12 47 17" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/64521658/a88d5d8c-6676-4c1e-8861-93078976a3b6">

vladyslavdobychin commented:
1. Adjusted Card component to account for the second status icon 
2. Made sure that the Sign Out button cannot be active
3. Re-arranged styling of the sidebar to match the proportions of already existing sidebar (Chat sidebar) 
4. Adjusted mobile styles to have same breakpoints and proportions with Chat sidebar

https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/fe562477-f534-4b9c-ab49-776be7d84e37

vladyslavdobychin commented:
Currently reviewing the changes and doing some additional testing

v0ldemar01 commented on frontend/src/libs/components/card/card.tsx:
```diff
@@ -27,24 +32,24 @@ const Card: React.FC<Properties> = ({
-        {title}
       </div>
+      <div className={styles['title']}>{title}</div>
+      {statusIcon && isActive && (
```
Create a boolean for whether existing statusIcon

v0ldemar01 commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-sidebar.tsx:
```diff
@@ -0,0 +1,83 @@
+
+import styles from './styles.module.scss';
+
+const settingsOptions: {
```
It is a constant, follow acceptance criteria for it. Also create a separate type for item

vladyslavdobychin replied:
I did add a type for for profile settings option, but not sure what exactly you meant by the acceptance criteria for it, I went over the code quality list and the closes I found was B22 - Words are separated by underscores (UPPER_SNAKE_CASE)

v0ldemar01 replied:
Yep, sure. But not resolved. And move it to separate file

vladyslavdobychin commented:
I think these are all the changes for now, if it's G2G I will remove the test buttons and extra icons.

v0ldemar01 commented on frontend/src/libs/components/card/card.tsx:
```diff
@@ -18,33 +21,41 @@ const Card: React.FC<Properties> = ({
-      </div>
+      {!hasNoImageOrIcon && (
+        <div className={styles['image-container']}>
+          {imageUrl && (
+            <div className={styles['image-placeholder']}>
+              <img src={imageUrl} alt="not found" className={styles['image']} />
+            </div>
+          )}
+          {iconName && (
+            <div className={styles['icon-background']}>
+              <Icon name={iconName} color={iconColor} />
+            </div>
+          )}
```
imageUrl  and iconName should be replaced with boolean bcos it would render it instead of jsx component

vladyslavdobychin replied:
That makes sense, working on it

v0ldemar01 reviewed (changes requested):
Everything is good, only one comment

liza-veis commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-sidebar.tsx:
```diff
@@ -0,0 +1,79 @@
+
+  const [activeButton, setActiveButton] = useState<IconName | null>(null);
+
+  const handleClick = useCallback((name: IconName) => {
+    return () => {
+      setActiveButton(name);
+    };
+  }, []);
```
let's remove this nested functions and use them like () => handleClick(name)
```suggestion
  const handleClick = useCallback((name: IconName) => {
    return setActiveButton(name);
  }, []);
```
```

vladyslavdobychin replied:
Initially I started with this Idea but linter was quite strict about it, so the nested function was the only solution I could figure out. 

<img width="733" alt="Screenshot 2023-09-10 at 18 13 59" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/201483e3-5258-495f-8523-75f4e4777816">

liza-veis replied:
ok, got it

liza-veis commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-sidebar.tsx:
```diff
@@ -0,0 +1,79 @@
+];
+
+const ProfileSettingsSidebar: React.FC = () => {
+  const { fullName } = useAppSelector(({ auth }) => {
+    return {
+      fullName: (auth.authenticatedUser as UserAuthResponseDto).fullName,
+    };
+  });
```
```suggestion
  const { authenticatedUser } = useAppSelector(({ auth }) => {
    return {
      authenticatedUser: (auth.authenticatedUser as UserAuthResponseDto),
    };
  });
```

liza-veis commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-sidebar.tsx:
```diff
@@ -0,0 +1,79 @@
+    };
+  });
+
+  const [activeButton, setActiveButton] = useState<IconName | null>(null);
```
rename it to activeItem and don't detect it by iconName

liza-veis commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-sidebar.tsx:
```diff
@@ -0,0 +1,79 @@
+  {
+    name: 'notification',
+    title: 'Very long text to see if it will break',
+    statusIcon: 'arrow',
```
let's remove this icon at all
```suggestion
```

vladyslavdobychin replied:
@Liza-Veis do you mean to remove it as a key/value pair from SettingsOption type or not to add the arrow Icon to buttons at all?

liza-veis commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-sidebar.tsx:
```diff
@@ -0,0 +1,79 @@
+
+const settingsOptions: SettingsOption[] = [
+  {
+    name: 'notification',
+    title: 'Very long text to see if it will break',
```
rename name to key, use actual value for title for notification item and remove subscription option

liza-veis commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-sidebar.tsx:
```diff
@@ -0,0 +1,79 @@
+              />
+            );
+          })}
+          <Card
```
should sign-out user on click

v0ldemar01 commented on frontend/src/slices/auth/actions.ts:
```diff
@@ -42,6 +42,17 @@ const signIn = createAsyncThunk<
+  async (_, { extra, dispatch }) => {
+    const { storage } = extra;
+    await storage.drop(StorageKey.TOKEN);
+    dispatch(appActions.navigate(AppRoute.SIGN_IN));
```
Does the private route already handle it?

vladyslavdobychin replied:
Are you saying that before adding the action I should have wrapped the sidebar in the protected route and inserted it in the index.tsx?

liza-veis replied:
I think @v0ldemar01 means that we don't need to navigate to sign_in here, because we will be automatically redirected when we drop authenticated user from store

v0ldemar01 replied:
As I know sign out functionality is available for auth user. Auth routes are wrapped with ProtectedRoute. That one has redirection depending on isAuthUser

vladyslavdobychin replied:
So basically, I do not need to explicitly define where to redirect the user since it will happen anyway? 
Okay, I do appreciate the clarification, thank you. 
I just wanted to make it visible what would happen in the end.

liza-veis replied:
yes

liza-veis replied:
user should be redirected to sign-in, but this should happen even without explicit redirect here

v0ldemar01 replied:
Sure)

vladyslavdobychin commented on frontend/src/slices/auth/auth.slice.ts:
```diff
@@ -54,6 +55,17 @@ const { reducer, actions, name } = createSlice({
       state.authDataStatus = DataStatus.REJECTED;
     });
 
```
I was not sure if SignOut action actually needs pending and rejected, but added them purely for the consistence purposes, to make sure signOut action follows the general pattern.

vladyslavdobychin commented on frontend/src/index.tsx:
```diff
@@ -85,6 +86,16 @@ createRoot(document.querySelector('#root') as HTMLElement).render(
+                element: (
+                  <ProtectedRoute>
+                    <SidebarWrapper>
+                      <ProfileSettingsSidebar />
```
Should I rename it into just 'Profile' or 'UserProfile'?

liza-veis replied:
should be the same as on mobile

vladyslavdobychin commented on frontend/src/libs/components/sidebar-wrapper/libs/constants.ts:
```diff
@@ -6,6 +6,7 @@ const SIDEBAR_ROUTES: Route[] = [
   { path: AppRoute.MEDITATION, name: 'meditation', icon: 'meditation' },
   { path: AppRoute.CHATS, name: 'chats', icon: 'chats' },
   { path: AppRoute.JOURNAL, name: 'journal', icon: 'journal' },
+  { path: AppRoute.PROFILE, name: 'profile', icon: 'avatar' },
```
Initially, the 'avatar' icon was used solely for displaying the placeholder for user's avatar, will rename it into something more meaningful in next commits. Like 'User'

vladyslavdobychin commented:
How it currently looks: 
Working on renaming icons. 

https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/2cca7b40-171d-42b0-890c-57bad6fb4f22

liza-veis commented on frontend/src/libs/components/card/card.tsx:
```diff
@@ -10,6 +11,8 @@ type Properties = {
   isActive: boolean;
   iconName?: IconName;
+  iconColor?: ValueOf<typeof IconColor>;
+  statusIcon?: IconName;
```
let's remove statusIcon
```suggestion
```

vladyslavdobychin replied:
@Liza-Veis just want to make sure we are on the same page: 
Are we talking about completely removing the Status Icon functionality from the Card component? 

<img width="400" alt="Screenshot 2023-09-11 at 16 05 25" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/09602de4-26e4-4809-a453-0ad78331b9c3">

liza-veis replied:
yes

liza-veis replied:
we will have an iconRight, but a bit later

liza-veis commented on frontend/src/libs/components/profile-settings-sidebar/styles.module.scss:
```diff
@@ -0,0 +1,77 @@
+  margin-right: 8px;
+  padding: 8px;
+  background-color: var(--blue-300);
+  border-radius: 9999px;
```
use %

vladyslavdobychin commented:
https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/ef1daae9-1f10-466d-83e6-4266e6dbed75

v0ldemar01 commented on frontend/src/libs/components/profile-settings-sidebar/libs/constants.ts:
```diff
@@ -0,0 +1,10 @@
+import { type SettingsOption } from '#libs/types/settings-options.type.js';
+
+const settingsOptions: SettingsOption[] = [
```
Follow quality criteria

vladyslavdobychin replied:
@v0ldemar01 just to clarify:
Are we talking about  B22. Constant names are written in capital letters ? 
I was not sure that I need to apply it in this case, since I was following general approach of chat sidebar that had its array with options in lowercase.

v0ldemar01 replied:
So it's a separate constant that need to be normalised with B22 rule

v0ldemar01 replied:
You can check other places of constants.ts to clarify about it

vladyslavdobychin replied:
Understood, working on it

v0ldemar01 commented on frontend/src/libs/components/profile-settings-sidebar/libs/constants.ts:
It should in folder constants

v0ldemar01 commented on frontend/src/libs/components/profile-settings-sidebar/profile-settings-sidebar.tsx:
What about file naming?

vladyslavdobychin replied:
My bad, fixing it as we speak

fshabanov commented on frontend/src/libs/components/profile-settings-sidebar/constants/constants.ts:
```diff
@@ -0,0 +1,10 @@
+import { type SettingsOption } from '#libs/types/settings-options.type.js';
```
```suggestion
import { type SettingsOption } from '#libs/types/types.js';
```

vladyslavdobychin replied:
You're right, my bad, working on it

fshabanov commented on frontend/src/libs/components/profile-settings-sidebar/user-profile.tsx:
```diff
@@ -0,0 +1,82 @@
+          <Card
+            title="Sign Out"
+            onClick={handleSignOut}
+            isActive={false}
```
Make `isActive` false by default and remove this line

fshabanov commented on frontend/src/libs/types/types.ts:
```diff
@@ -9,6 +9,7 @@ export {
 export { type IconName } from './icon-name.type.js';
 export { type Meditation } from './meditation-player/meditation-player.type.js';
 export { type Route } from './route/route.type.js';
+export { type SettingsOption } from './settings-options.type.js';
```
```suggestion
export { type SettingsOption } from './settings-option.type.js';
```

v0ldemar01 commented on frontend/src/libs/components/profile-settings-sidebar/constants/constants.ts:
Where is libs/

vladyslavdobychin replied:
Well, I renamed libs to constants, looks like it must be libs/constants right?

vladyslavdobychin replied:
Good morning, @v0ldemar01 
Previously you advised to put the constants file in constant folder which I did, you also left a comment about missing libs/ 
I tried checking other components to see how it is structured in the already approved components. It's either libs/constants.ts or it's component/constants/constants.ts 
Could you clarify what would be the right one? Thanks. 

<img width="819" alt="Screenshot 2023-09-12 at 10 30 33" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/204cfc8e-6de9-48f2-9177-4ddad7d362e1">

<img width="272" alt="Screenshot 2023-09-12 at 10 35 24" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/c597b382-1e7b-4db7-9f5f-26a58a7d6747">

<img width="311" alt="Screenshot 2023-09-12 at 10 35 53" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/f082e0ae-2daa-4a05-99b9-9aa17a9106de">

liza-veis replied:
let's use `libs/constants.ts` for now as it was done initially in the template, we will replace it with `libs/constants/constants.ts` later

liza-veis commented on frontend/src/libs/components/card/card.tsx:
```diff
@@ -1,50 +1,56 @@
+  isActive?: boolean;
   iconName?: IconName;
+  iconColor?: ValueOf<typeof IconColor>;
+  statusIcon?: IconName;
```
remove statusIcon property

liza-veis commented on frontend/src/slices/auth/actions.ts:
```diff
@@ -42,6 +42,16 @@ const signIn = createAsyncThunk<
+    const { storage } = extra;
+    await storage.drop(StorageKey.TOKEN);
+
+    return null;
```
we don't need to return anything here
```suggestion
```

vladyslavdobychin replied:
Initially, I wanted to do so, but without any return statement I get an error. Couldn't find a right solution for it. 
So I tried following the same approach that followed every other action - return something even if it's null: 
<img width="323" alt="Screenshot 2023-09-11 at 20 55 35" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/287b891d-f2d3-418e-ba48-2493259404fd">
<img width="329" alt="Screenshot 2023-09-11 at 20 55 47" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/c978a990-1ade-41a9-b873-b8594a4efaea">
<img width="776" alt="Screenshot 2023-09-11 at 20 55 57" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/878fe2aa-feda-4cc5-80f3-0b1f707e1cfa">

liza-veis replied:
pass unknown instead of null as a first parameter of generic

vladyslavdobychin replied:
I see, thank you

Liza-Veis reviewed (changes requested):
everything looks good to me, fix two comment above

vladyslavdobychin commented:
@Liza-Veis both comments fixed as requested 

https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/1318cb0f-4faf-41be-8218-750f2665e1f1
